parlai/core/torch_generator_agent.py

```diff
@@ -849,6 +874,17 @@ def __init__(
         self.n_best_counter = 0
         self.partial_hyps = [[self.bos] for i in range(beam_size)]
 
+    def set_context(self: TSType, context: torch.LongTensor) -> TSType:
```

Comment:
does this typing raise an error if `context` doesn't have the correct type?

Reply:
this would, yes. the tolist would fail.

Reply:
but the typechecker is never used at runtime